File: src/main/java/com/paragon/client/systems/ui/alt/AltEntry.java
package com.paragon.client.systems.ui.alt;

import com.paragon.api.util.Wrapper;
import com.paragon.api.util.render.RenderUtil;
import com.paragon.api.util.render.TextRenderer;
import com.paragon.client.managers.alt.Alt;

public class AltEntry implements TextRenderer, Wrapper {

    private Alt alt;
    private float offset;

    public AltEntry(Alt alt, float offset) {
        this.alt = alt;
        this.offset = offset;
    }

    public void drawAlt(int mouseX, int mouseY, int screenWidth) {
        RenderUtil.drawRect(0, offset, screenWidth, 20, AltManagerGUI.selectedAltEntry == this ? 0x95111111 : 0x95000000);
        renderCenteredString(alt.getEmail(), screenWidth / 2f, offset + 10, -1, true);
    }

    public void clicked(int mouseX, int mouseY, int screenWidth) {
        if (isHovered(0, offset, screenWidth, 20, mouseX, mouseY)) {
            alt.login();
        }
    }

    public float getOffset() {
        return offset;
    }

    public void setOffset(float newOffset) {
        this.offset = newOffset;
    }

    public Alt getAlt() {
        return alt;
    }

}

File: src/main/kotlin/com/paragon/impl/ui/alt/AltEntry.kt
package com.paragon.impl.ui.alt

import com.paragon.util.render.font.FontUtil.drawCenteredString
import com.paragon.impl.managers.alt.Alt
import com.paragon.util.Wrapper
import com.paragon.util.render.RenderUtil.drawRect
import java.awt.Color

class AltEntry(val alt: Alt, var offset: Float) : Wrapper {

    fun drawAlt(screenWidth: Int) {
        drawRect(
            0f,
            offset,
            screenWidth.toFloat(),
            20f,
            if (AltManagerGUI.selectedAltEntry === this) Color(238, 238, 239, 150) else Color(0, 0, 0, 150)
        )

        drawCenteredString(alt.email, screenWidth / 2f, offset + 10, Color.WHITE, true)
    }

    fun clicked(mouseX: Int, mouseY: Int, screenWidth: Int) {
        if (isHovered(0f, offset, screenWidth.toFloat(), 20f, mouseX, mouseY)) {
            alt.login()
        }
    }

}